<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>SRA-toolkit Tutorial - My GitHub Portfolio</title>
    <link rel="stylesheet" href="../style.css">
    <link href="https://fonts.googleapis.com/css2?family=Poppins:wght@300;400;600&display=swap" rel="stylesheet">
</head>
<body>
    <header>
        <div class="header-container">
            <h1>SRA-toolkit Tutorial</h1>
            <nav>
                <ul>
                    <li><a href="../index.html">Home</a></li>
                    <li><a href="../introduction.html">Introduction</a></li>
                    <li><a href="../tutorials.html">Tutorials</a></li>
                    <li><a href="../publications.html">Publications</a></li>
                    <li><a href="../blogs.html">Blogs</a></li>
                    <li><a href="../bioinformatics-news.html">Bioinformatics News</a></li>
                </ul>
            </nav>
        </div>
    </header>

    <main>
        <section class="tutorial-content">
            <h2>SRA-toolkit Tutorial</h2>
            <p><strong>What is SRA-toolkit?</strong></p>
            <p>The SRA-toolkit is a set of utilities to download and process sequencing data from the NCBI Sequence Read Archive (SRA) at scale. The SRA is a primary repository for high-throughput sequencing data hosted by NIH and is part of the International Nucleotide Sequence Database Collaboration (INSDC).</p>

            <h3>Downloading the SRA-toolkit</h3>
            <p>Download the latest version of the SRA-toolkit (version 3.0.0) using the following command:</p>
            <pre><code>wget https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/3.0.0/sratoolkit.3.0.0-ubuntu64.tar.gz</code></pre>

            <h3>Extracting the Toolkit</h3>
            <p>Extract the downloaded tar.gz file:</p>
            <pre><code>tar -zxvf sratoolkit.3.0.0-ubuntu64.tar.gz</code></pre>

            <h3>Configuring the SRA-toolkit</h3>
            <p>Add the SRA-toolkit binaries to your PATH:</p>
            <pre><code>export PATH=$PATH:/home/Abbas/tools/sratoolkit.3.0.0-ubuntu64/bin</code></pre>
            <p>Configure the SRA-toolkit to access public cloud data:</p>
            <pre><code>vdb-config -i</code></pre>
            <p>Follow the on-screen instructions to configure the toolkit. For more details, visit the <a href="https://github.com/ncbi/sra-tools/wiki/03.-Quick-Toolkit-Configuration" target="_blank">SRA-toolkit configuration guide</a>.</p>

            <h3>Downloading Sequencing Data</h3>
            <p>To download a single SRA file:</p>
            <pre><code>prefetch SRR19850882</code></pre>
            <p>To download multiple SRA files:</p>
            <pre><code>prefetch SRR19850882 SRR19850883 SRR19850884</code></pre>
            <p>You can also provide a text file containing SRR numbers to download multiple files.</p>

            <h3>Converting SRA Files to FASTQ Format</h3>
            <p>Use <code>fastq-dump</code> or <code>fasterq-dump</code> to convert SRA files to FASTQ format:</p>
            <pre><code>fastq-dump SRR19850882 SRR19850883 SRR19850884</code></pre>
            <p>For paired-end sequencing data, use the <code>--split-files</code> option:</p>
            <pre><code>fastq-dump --split-files SRR19850882 SRR19850883 SRR19850884</code></pre>
            <p>Alternatively, use <code>fasterq-dump</code> for faster processing:</p>
            <pre><code>fasterq-dump SRR19850882 SRR19850883</code></pre>

            <h3>Additional Resources</h3>
            <p>For more advanced usage, check out the <a href="https://github.com/rvalieris/parallel-fastq-dump" target="_blank">parallel-fastq-dump</a> tool.</p>
        </section>
    </main>

    <footer>
        <p>Â© 2025 Mubashir3164 | <a href="mailto:mubashirabbas3164@yahoo.com">Contact Me</a></p>
    </footer>
</body>
</html>
